### PR TITLE
Devex 851 case helper 1.8

### PR DESCRIPTION
**Pull request by kkoskelin**
Hoist up from persistence method to helper:
* updates to irs practitioners
* updates to private practitioners

Some new persistence methods dedicated to these purposes allow us to check the method is called, rather than inspecting the arguments passed to the persistence client itself.

### CODE REVIEW

File: shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js [hunk 1]
```diff
@@ -277,4 +277,194 @@ describe('updateCaseAndAssociations', () => {
...
+      });
+      expect(
+        applicationContext.getPersistenceGateway().updateIrsPractitionerOnCase,
+      ).not.toHaveBeenCalled();
+      expect(
+        applicationContext.getPersistenceGateway().removeIrsPractitionerOnCase,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('calls updateIrsPractitionerOnCase on changed entries in irsPractitioners', async () => {
```
Comment (rachelschneiderman) [inline shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js:317]: This test is very well written!

File: shared/src/persistence/dynamo/cases/updatePractitionerOnCase.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const client = require('../../dynamodbClientService');
+
+exports.updateIrsPractitionerOnCase = async ({
```
Comment (rachelschneiderman) [inline shared/src/persistence/dynamo/cases/updatePractitionerOnCase.js:3]: General question, I see some of the persistence methods have unit tests and others do not. Is there an agreement on whether or not these files should be unit tested?

**Review by rachelschneiderman — APPROVED** [review]

File: shared/src/persistence/dynamo/cases/updatePractitionerOnCase.js (hunk 2 above)
Comment (kkoskelin) [reply shared/src/persistence/dynamo/cases/updatePractitionerOnCase.js:3]: Great observation -- tbh it was unclear to me whether there was value in testing precisely what was being sent to the client.  I'm open to writing coverage for these if we think it has value.

Comment (rachelschneiderman) [reply shared/src/persistence/dynamo/cases/updatePractitionerOnCase.js:3]: I think there's value in unit testing these but I won't hold up the PR for it!

**Pull request merged**